qt-bugs@ issue : none
Trolltech task ID : none
bugs.kde.org number : None
applied: yes
author: Trolltech ASA

Fix QUrl::clear().

--- src/corelib/io/qurl.cpp	Tue May 29 14:55:12 CEST 2007
+++ src/corelib/io/qurl.cpp	Tue May 29 14:55:12 CEST 2007

@@ -3411,6 +3411,7 @@
     fragment.clear();
 
     encodedOriginal.clear();
+    encodedNormalized.clear();
 
     isValid = false;
     hasQuery = false;
@@ -3419,7 +3420,7 @@
     valueDelimiter = '=';
     pairDelimiter = '&';
 
-    QURL_UNSETFLAG(stateFlags, Parsed | Validated);
+    QURL_UNSETFLAG(stateFlags, Parsed | Validated | Normalized);
 }
 
 QByteArray QUrlPrivate::toEncoded(QUrl::FormattingOptions options) const

